packages/hierarchies-react/src/presentation-hierarchies-react/UseNodeHighlighting.tsx

```diff
@@ -0,0 +1,217 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { useCallback, useEffect, useState } from "react";
+import { PresentationHierarchyNode } from "./TreeNode.js";
+import { useLatest } from "./Utils.js";
+
+/** @alpha */
+export interface HighlightInfo {
+  text: string;
+}
+
+/**
+ * Props for `useNodeHighlighting` hook.
+ * @alpha
+ */
+interface UseNodeHighlightingProps {
+  rootNodes: PresentationHierarchyNode[] | undefined;
+  // TODO: move activeMatchIndex and onHighlightChanged to HighlightInfo when it's implemented.
+  highlight?: HighlightInfo & {
+    activeMatchIndex?: number;
+    onHighlightChanged?: (activeMatchIndex: number, matches: number) => void;
+  };
```

Comment:
Is there value in having this `HighlightInfo` type and `highlight` attribute? It seems we could just move all these props straight into `UseNodeHighlightingProps`